pkg/webui/console/views/device/index.js

```diff
@@ -0,0 +1,117 @@
+// limitations under the License.
+
+import React from 'react'
+import { connect } from 'react-redux'
+import { Switch, Route } from 'react-router'
+import { Helmet } from 'react-helmet'
+import { Col, Row, Container } from 'react-grid-system'
+
+import sharedMessages from '../../../lib/shared-messages'
+import Message from '../../../lib/components/message'
+import { withBreadcrumb } from '../../../components/breadcrumbs/context'
+import Breadcrumb from '../../../components/breadcrumbs/breadcrumb'
+import Spinner from '../../../components/spinner'
+import Tabs from '../../../components/tabs'
+
+import DeviceOverview from '../device-overview'
+
+import { getDevice } from '../../store/actions/device'
+
+import style from './device.styl'
+
+const tabs = [
+  { title: sharedMessages.overview, name: 'overview' },
+  { title: sharedMessages.data, name: 'data' },
+  { title: sharedMessages.location, name: 'location' },
+  { title: sharedMessages.payloadFormats, name: 'develop' },
+  { title: sharedMessages.generalSettings, name: 'general-settings' },
+]
+
+@connect(function ({ device, application }, props) {
+  return {
+    appName: application.application.name,
+    devId: props.match.params.devId,
+    fetching: device.fetching,
+    error: device.error,
+  }
+})
+@withBreadcrumb('device.single', function (props) {
+  const { devId } = props
+  const { appId } = props.match.params
+  return (
+    <Breadcrumb
+      path={`/console/applications/${appId}/devices/${devId}`}
+      icon="device"
+      content={devId}
+    />
+  )
+})
+export default class Device extends React.Component {
+
+  componentDidMount () {
+    const { dispatch, devId, match } = this.props
+    const { appId } = match.params
+
+    dispatch(getDevice(appId, devId, 'name,description,session,version_ids', { ignoreNotFound: true }))
+  }
+
+  handleTabChange () {
+
+  }
+
+  render () {
+    const { fetching, error, match, appName, devId } = this.props
+    const { appId } = this.props.match.params
+
+    if (fetching) {
+      return (
+        <Spinner center>
+          <Message content={sharedMessages.loading} />
+        </Spinner>
+      )
+    }
+
+    // show any device fetching error, e.g. not found, no rights, etc
+    if (error) {
+      return 'ERROR'
+    }
+
+    return (
+      <React.Fragment>
+        <Helmet
+          titleTemplate={`%s - ${appName || appId} - The Things Network Console`}
+          defaultTitle="Devices - The Things Network Console"
+        />
+        <Container>
+          <Row>
+            <Col lg={12}>
+              <h2 className={style.title}>{devId}</h2>
+              <Tabs
+                narrow
+                active="overview"
+                tabs={tabs}
+                onTabChange={this.handleTabChange}
+                className={style.tabs}
+              />
+            </Col>
+          </Row>
+        </Container>
+        <Switch>
+          <Route exact path={`${match.path}`} component={DeviceOverview} />
```

Comment:
Do you really want to render tabs on every device-related screen? add/edit/overview? this seems weird.

Reply:
Yeah, this needs to be refactored a little to work like we do it with applications (having a `device` and `devices` route). Will do.